libs-gui/Documentation/set_show_service.1
.\"set_show_service(1) man page
.\"put together by Dennis Leeuw <dleeuw@made-it.com>
.\"Copyright (C) 2003 Free Software Foundation, Inc.
.\"
.\"Process this file with
.\"groff -man -Tascii set_show_service.1
.\"
.TH SET_SHOW_SERVICE 1 "December 2007" GNUstep "GNUstep System Manual"
.SH NAME
set_show_service \- enable or disable a service for the current user
.SH SYNOPSIS
.B set_show_service [--enable <service>] [--disable <service>]
.P
.SH DESCRIPTION
.B set_show_service
enables or disables the display of a specified service item.
.SH OPTIONS
.TP
\fB\--enable\fR \fB\<service>\fR
enable the given service for the user running this command.

.TP
\fB\--disable\fR
disables the given service for the user running this command.

.SH BUGS
None known

.P
.SH SEE ALSO
make_services(1), GNUstep(7)
.P
.SH HISTORY
Work on
.B set_show_service
started in November 1998.
.P
.SH AUTHORS
set_show_service was written by Richard Frith-Macdonald <richard@brainstorm.co.uk>
.P
This man page was written by Dennis Leeuw <dleeuw@made-it.com>.


libs-gui/NEWS
1 NEWS
******

The currently released version of the library is '0.28.0'.

1.1 Noteworthy changes in version '0.28.0'
==========================================

This version adds support for modern XIB files and many new classes.
Plus the usual bunch of bug fixes.

   * Support loading of document XIB files.
   * Improve Key Value Binding for NSArrayController and add more
     bindings.
   * Better support for multi monitor usage and other improvement in the
     backend integration.
   * Add classes NSFontCollection, NSColorSampler, NSSpeechRecognizer,
     NSAppearance, NSPDFInfo, NSPICTImageRep, NSCIImageRep,
     NSPDFImageRep, NSPDFPanel, NSDataAsset, NSDatePicker,
     NSDatePickerCell, NSPredicateEditor, NSPredicateEditorRowTemplate,
     NSRuleEditor, NSGestureRecognizer, NSButtonTouchBarItem,
     NSCandidateListTouchBarItem, NSClickGestureRecognizer,
     NSColorPickerTouchBarItem, NSCustomTouchBarItem,
     NSGroupTouchBarItem, NSMagnificationGestureRecognizer,
     NSPanGestureRecognizer, NSPickerTouchBarItem,
     NSPopoverTouchBarItem, NSPressGestureRecognizer,
     NSRotationGestureRecognizer, NSSharingServicePickerTouchBarItem,
     NSSliderTouchBarItem, NSStepperTouchBarItem, NSTouchBarItem,
     NSTouchBar, NSTouch, NSDockTile.
   * Implement NSEPSImageRep.
   * Better encoding handling in RTF files.
   * Theming and drawing improvements.
   * Increase small font size to 10.
   * New cursor and stepper images.
   * Move NSFileWrapper to Foundation.
   * Fixed build on Debian GNU/kFreeBSD.
   * With command line argument -autolaunch YES, do not activate the
     application when -activateIgnoringOtherApps: is invoked.
   * Improvements to WindowMaker compatibility (e.g.  WMFHideApplication
     support).
   * Lowered NSFloatingWindowLevel by one to distinguish floating panels
     from menus.

1.2 Noteworthy changes in version '0.27.0'
==========================================

This version includes numerous bugfixes, compatibility improvements and
other changes accumulated over the last year.  It also enables work to
be done on integrating NSViews with a Core Animation renderer.

   * Make targetForAction safer.
   * Speed up menu updates.
   * Clean up speech tool compilation and switch to newer interface.
   * Fix bug in CUPS subclassing introduced in last release.
   * Minor improvements to typesetting.
   * Add NSIsControllerMarker.
   * Fix tracking on segmented cell.
   * Bring slider cell closer to Cocoa implementation.
   * Add ivar for Core Animation in NSView.
   * Improve border calculation on printing.
   * Lazy load app icon.
   * Better detection of removable volumes.
   * Polish translations.
   * Japanese translations.
   * Lots of bug fixes.

1.3 Noteworthy changes in version '0.26.2'
==========================================

This version is a small, but important bugfix release.

   * printing: Fix allocation of the CUPS printing classes.
   * installation: Fix the configure script.

1.4 Noteworthy changes in version '0.26.1'
==========================================

This version is released to conincide with version 1.25.1 of
gnustep-base, which contains changes required for this version of
gnustep-gui and gnustep-back.

   It includes an important workaround for users of GNUstep Objective-C
Runtime (libobjc2) and non-fragile ABI to avoid a bug in interaction
between the clang compiler and the runtime when non-fragile ABI is in
use.  Specifically, Clang and the runtime may disagree on what is the
offset of an ivar in a class's RAM. This manifested in a crash at
application startup due to misalignment of _gcontext inside NSThread.
See the mailing list discussion
(http://lists.gnu.org/archive/html/discuss-gnustep/2017-12/msg00129.html)
for more information.

   It also contains the following changes:

   * tests: Cleanup of warnings.
   * tests: Fix text system deallocation test.
   * printing: Undefine __BLOCKS__ before including cups.h, as some
     versions of the header expect that libdispatch is present and used
     if __BLOCKS__ is defined.
   * graphics context: Workaround for Clang+libobjc2+nonfragile ABI
     issue.

1.5 Noteworthy changes in version '0.26.0'
==========================================

This version was bumped due to previous binary incompatibilities between
0.25.0 and 0.25.1.  This version also includes numerous compatibility
improvements from the Summer of Code project, and a wide variety of
other fixes.  Notably, it fixes the use of cupsGetPPD() in the printing
system.

   * printing: Add an include to get deprecated function cupsGetPPD() on
     newer CUPS systems.
   * chore: Bump required base version.
   * tiff: Support for writing resolution.
   * jpeg: Save resolution information if it is different from 72 dpi.
   * save panel: Fix return type of sorting function.
   * events: Add some newer Cocoa enums and one method with dummy
     implementation.
   * speech synthesis: NSSpeechSynthesizerDelegate is now a @protocol on
     runtimes that support it.
   * pasteboard: New type identifiers.
   * translations: Some work on Polish, Russian and German translations
   * cell: Improvements to mouse tracking logic on NSCell.
   * image: If an unknown named image is unarchived with a coder or
     keyed coder, keep the name.
   * screen: Add -backingScaleFactor and return 1.0.
   * window: Return 1.0 from -backingScaleFactor.

   * compatibility: Numerous stub implementations of constants, classes
     and methods to improve source-level compatibility.
   * other bugfixes

1.6 Noteworthy changes in version '0.25.1'
==========================================

   * JPEG (saving) alpha channel fixes and size with resolution != 72
   * JPEG resolution read support
   * TIFF saving fixes
   * Improved volumes mounting and support
   * Portability improvements in volume mounting and support
   * Corrected layout of empty strings
   * Only update visible menus

1.7 Noteworthy changes in version '0.25.0'

libs-gui/Tools/set_show_service.m
/* 
   set_show_service.m

   GNUstep utility to enable or disable a service for the current user.

   Copyright (C) 1998 Free Software Foundation, Inc.

   Author:  Richard Frith-Macdonald <richard@brainstorm.co.uk>
   Date: November 1998
   
   This file is part of the GNUstep Project

   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 3
   of the License, or (at your option) any later version.
    
   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public  
   License along with this library; see the file COPYING.
   If not, see <http://www.gnu.org/licenses/> or write to the 
   Free Software Foundation, 51 Franklin Street, Fifth Floor, 
   Boston, MA 02110-1301, USA.

*/ 

#include <stdlib.h>
#import <Foundation/NSAutoreleasePool.h>
#import <Foundation/NSArray.h>
#import <Foundation/NSProcessInfo.h>
#import <Foundation/NSString.h>
#import "AppKit/NSApplication.h"

int
main(int argc, char** argv, char **env)
{
  NSAutoreleasePool	*pool;
  NSProcessInfo		*proc;
  NSArray		*args;
  unsigned		index;

  // [NSObject enableDoubleReleaseCheck: YES];
#ifdef GS_PASS_ARGUMENTS
  [NSProcessInfo initializeWithArguments:argv count:argc environment:env];
#endif

  pool = [NSAutoreleasePool new];

  proc = [NSProcessInfo processInfo];
  if (proc == nil)
    {
      NSLog(@"unable to get process information!\n");
      exit(EXIT_SUCCESS);
    }

  args = [proc arguments];

  for (index = 1; index < [args count]; index++)
    {
      if ([[args objectAtIndex: index] isEqual: @"--help"])
	{
	  printf(
"set_show_service enables or disables the display of a specified service\n"
"item.  It's should be in the form 'set_show_service --enable name' or \n"
"'set_show_service --disable name' where 'name' is a service name.\n");
	  exit(EXIT_SUCCESS);
	}
      if ([[args objectAtIndex: index] isEqual: @"--enable"])
	{
	  if (index >= [args count] - 1)
	    {
	      NSLog(@"No name specified for enable.\n");
	      exit(EXIT_FAILURE);
	    }
	  NSSetShowsServicesMenuItem([args objectAtIndex: ++index], YES);
	  exit(EXIT_SUCCESS);
	}
      if ([[args objectAtIndex: index] isEqual: @"--disable"])
	{
	  if (index >= [args count] - 1)
	    {
	      NSLog(@"No name specified for disable.\n");
	      exit(EXIT_FAILURE);
	    }
	  NSSetShowsServicesMenuItem([args objectAtIndex: ++index], NO);
	  exit(EXIT_SUCCESS);
	}
    }

  NSLog(@"Nothing to do.\n");
  [pool drain];
  return(1);
}

